### DISCUSSION TITLE
How to import the key value pair from a file to apollo server

### DISCUSSION THREAD

**Discussion by jyotibhanot** [apolloconfig/apollo#4902] [Q&A]
I have a test.properties file that has some data like:

key1 : value1
key2 : value2

How can I import these values in apollo using curl commands?

**Comment by nobodyiam** [comment]
You may use the import namespace feature. If you insist on using the curl command, then you need to get the session id and simulate the session information via cookie.

![image](https://github.com/apolloconfig/apollo/assets/837658/0faa8249-6ba5-41d8-b6f9-1b346eb51f3c)

**Comment by jyotibhanot** [comment]
What will be the curl to post a key value pair to grayscale? I am unable to find in documentation: https://www.apolloconfig.com/#/en/usage/apollo-open-api-platform?id=_3210-new-configuration-interface

**Reply by nobodyiam** [reply]
You may reference #1518 for more information. The grayscale API is complicated, so it is not explicitly mentioned in the documentation.

**Reply by jyotibhanot** [reply]
What will be the equivalent curl for it? 

curl --header "Authorization: xxx" --header "Content-Type: application/json;charset=UTF-8" "https://axxx/openapi/v1/envs/QA/apps/test/clusters/default/namespaces/application/branches"

does not give me the branches

**Reply by nobodyiam** [reply]
You may check the [NamespaceBranchController](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/NamespaceBranchController.java).
